### PR TITLE
carbon_black_cloud: ensure alert search range is a valid temporal ordering

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/carbon_black_cloud/data_stream/alert_v7/agent/stream/cel.yml.hbs [hunk 1]
```diff
@@ -31,10 +30,18 @@ program: |-
...
-  				"time_range": {
-  					"start": state.?cursor.last_backend_update_timestamp.orValue(string(now - duration(state.initial_interval) + duration("-15m"))),
-  					"end": string(now + duration("-15m")),
-  				},
+          // The alert data stream has a 15-minute delay to ensure that no
+          // occurrences are missed. It is possible that we may have received
+          // an event that is after this delay for the next evaluation. This
+          // would result in an inverted time range. So clamp the range start
+          // to the last event we received.
+  				"time_range": (now - duration("15m")).as(delayed, {
```
Comment (chrisberkhout) [inline packages/carbon_black_cloud/data_stream/alert_v7/agent/stream/cel.yml.hbs:38]: This logic looks good.

As mentioned elsewhere, I think we can reduce the 15m to 60s, remove the [note](https://github.com/elastic/integrations/blob/2e72e8524728daca2d47c814d8042031b8f5804f/packages/carbon_black_cloud/_dev/build/docs/README.md?plain=1#L81-L83) in the README and link to the [documentation](https://developer.carbonblack.com/reference/carbon-black-cloud/guides/alert-bulk-export/) here.

**Review by chrisberkhout — CHANGES_REQUESTED** [review]

File: packages/carbon_black_cloud/data_stream/alert_v7/agent/stream/cel.yml.hbs (hunk 1 above)
Comment (efd6) [reply packages/carbon_black_cloud/data_stream/alert_v7/agent/stream/cel.yml.hbs:38]: Do you think it's worth making this configurable?

Comment (chrisberkhout) [reply packages/carbon_black_cloud/data_stream/alert_v7/agent/stream/cel.yml.hbs]: No, I don't think so.

I think it's probably just a matter of a timestamp being assigned a value in application code and then written after a variable delay, or being assigned in a database with multiple write nodes and then replicated with variable delay. So they're out of order by roughly the duration of a request. There could be other systems with longer delays, but if they say 60 seconds it's probably 1ms - 15s that we're really worried about.

**Review by chrisberkhout — APPROVED** [review]

**Comment by jamiehynds** [comment]
@efd6 there's a custom on our Discuss forums still reporting an issue with Carbon Black after updating to the latest integration. https://discuss.elastic.co/t/carbon-black-cloud-cel-alert-v7-400-bad-request/366464/3

**Pull request merged**